mempool/reactor.go

```diff
@@ -224,25 +225,26 @@ func (memR *Reactor) broadcastTxRoutine(peer p2p.Peer) {
 			time.Sleep(peerCatchupSleepIntervalMS * time.Millisecond)
 			continue
 		}
+
+		memTx := next.Value.(*mempoolTx)
 		if peerState.GetHeight() < memTx.Height()-1 { // Allow for a lag of 1 block
 			time.Sleep(peerCatchupSleepIntervalMS * time.Millisecond)
 			continue
 		}
```

Comment:
Could this logic, as well as some of the duplicated logic inside `txs()`, be combined and generalized? It's a bit hard to reason about when the logic is special-cased for the first tx and split up like this.

Reply:
I will try to simplify